client/src/App.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import styles from './App.module.css';
+import API from './constant/api';
+import FlightLists from './components/Flightlists/Flightlists';
+import Layout from './components/Layout/Layout';
+import Sidefilter from './components/Sidefilter/Sidefilter';
+import axios from 'axios';
+
+const App = () => {
+  const [filteredData, setFilteredData] = useState([]);
+  const [returnData, setReturnData] = useState([]);
+  const [isLoading, setIsloading] = useState(false);
+  const [isError, setIserror] = useState(false);
+  const [isErrorMessage, setIserrorMessage] = useState('');
+  const [showReturnData, setshowReturnData] = useState(false);
+  const [places, setPlaces] = useState([]);
+
+  const onSearch = (event, formData) => {
+    event.preventDefault();
+    onSearchCall(formData);
+  };
+
+  const onSearchCall = async (formData) => {
+    try {
+      const response = await axios.post(`${API.serverapi}/search`, {
+        returnDataFlag: showReturnData,
+        formData,
+      });
+      setFilteredData(response.data.flightOneWayData);
+      if (showReturnData) {
```

Comment:
add if (response?.data?.returnFlightData)). 
this will prevent your code from errors in api failure